lib/vagrant-libvirt/util/network_util.rb

```diff
@@ -59,6 +59,11 @@ def configured_networks(env, logger)
               dhcp_enabled: true,
               forward_mode: 'nat',
             }.merge(options)
+
+            if options[:type] == :dhcp && !options[:ip]
```

Comment:
why? the nil check is only useful if the variable may contains boolean value, which is not the case here. A truthy check is enough and is more conventional ruby.